Tutorial-Learn-how-to-make-a-backend-change-that-has-an-impact-on-server-data.md

```diff
@@ -0,0 +1,677 @@
+self.assertNotEqual(updated_user_2.last_updated, original_last_updated_2)
+```
+
+***Note:** In addition to testing the migration job, it is important to test the audit job (`AuditTruncateUserBioJob`) to ensure that it correctly identifies records needing truncation without making any changes to the datastore. While the implementation of audit job tests is not shown here, it follows a similar structure, focusing on validating read-only operations and accurate reporting.*
+
+## Section 5: Run and Validate the Job
+
+Once the `TruncateUserBioJob` has been written and tested, the next step is to run the job on a local server and validate its behavior. Running the job allows you to see how it interacts with real data and confirm that it performs as expected. This section walks you through the steps to run the job and test it with various scenarios.
+
+Now let’s try running the job on our local server. 
+
+1. Sign in as an administrator ([instructions](https://github.com/oppia/oppia/wiki/How-to-access-Oppia-webpages#log-in-as-a-super-administrator)).  
+2. Navigate to Admin Page \> Roles Tab.  
+3. Add the "Release Coordinator" role to the username you are signed in with.  
+4. Navigate to [http://localhost:8181/release-coordinator](http://localhost:8181/release-coordinator), then to the Beam Jobs tab.  
+5. Search for your job by name (e.g., `TruncateUserBioJob`).  
+6. Click the **Play** button next to the job name.  
+7. Click **Start a New Job** to begin execution.
+
+To thoroughly validate the job, you’ll need to test it under various conditions. Follow these steps to create dummy data and observe the job’s behavior:
+
+1. **Initial Run Without Dummy Data**  
+   * Without creating any dummy data, run the job.  
+   * Since no data exists in the datastore, the behavior will match the "Null Case" from the test suite.  
+   * **Expected Outcome**: The job should complete without making any changes, and the output should indicate that no models were processed.  
+2. **Populate the Datastore with Users**  
+   * Sign up as different users to create new entries in the datastore.  
+   * Visit the Preferences page ([http://localhost:8181/preferences](http://localhost:8181/preferences)) for each user and add data to the `Bio` field.  
+3. **Create Various Test Cases**  
+   * For some users, set the `user_bio` field to within the 200-character limit.  
+   * For others, enter a `user_bio` that exceeds 200 characters.  
+   * Leave the `user_bio` field empty for additional cases.  
+4. **Run the Job on Populated Data**  
+   * Navigate back to the **Release Coordinator** page and start the job again.  
+   * Observe the behavior and verify whether it matches the expected outcomes for each scenario:  
+     * **Bios within the limit**: Remain unchanged.  
+     * **Bios exceeding the limit**: Are truncated to 200 characters.  
+     * **Empty bios**: Remain unaffected.
+
+### **Conclusion**
```

Comment:
Can you explain, somewhere in this tutorial, the "sequence of operations"? I.e. they need to do the domain-layer fix to stop new bios > 200 chars getting added. When do they make this fix, when do they run the job on the server, when are things deployed, etc. -- what is the sequence of operations?

I think this is something that probably needs to be explained at the outset of the tutorial, because fixing an issue that affects existing data is an entire process and that process needs to happen in the right order.